controllers/clustergroupupgrade_controller.go

```diff
@@ -151,21 +151,25 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	// Verify that the selected clusters are all valid
 	// If this has already been done we will skip it on further reconcile loops
+
```

Comment:
Shouldn't this whole block be in the block under the progressing condition check?

Reply:
No, since I am using validateCR to generate the initial cluster list and using it again below I need it to run every time

Reply:
That block is run every time until progressing becomes true. We don't want to re-select clusters if it's already in progress.